lib/gems/pending/util/object_storage/miq_swift_storage.rb

```diff
@@ -0,0 +1,148 @@
+      msg = "Access to Swift container #{@container_name} failed due to a bad username or password. #{err}"
+      raise err, msg, err.backtrace
+    rescue => err
+      logger.error("Error uploading #{source_input} to Swift container #{@container_name}. #{err}")
+      msg = "Error uploading #{source_input} to Swift container #{@container_name}. #{err}"
+      raise err, msg, err.backtrace
+    end
+  end
+
+  def mkdir(_dir)
+    container
+  end
+
+  def container
+    @container ||= begin
+                     container   = swift.directories.get(container_name)
+                     logger.debug("Swift container [#{container}] found") if container
+                     container ||= create_container
+                     container
+                   rescue Fog::Storage::OpenStack::NotFound
+                     logger.debug("Swift container #{container_name} does not exist.  Creating.")
+                     create_container
+                   rescue => err
+                     logger.error("Error getting Swift container #{container_name}. #{err}")
+                     msg = "Error getting Swift container #{container_name}. #{err}"
+                     raise err, msg, err.backtrace
+                   end
+  end
+
+  private
+
+  def swift
+    return @swift if @swift
+    require 'manageiq/providers/openstack/legacy/openstack_handle'
+    extra_options = {}
+    extra_options[:domain_id] = @domain_id
+    extra_options[:service] = "Compute"
+     @osh ||= OpenstackHandle::Handle.new(@username, @password, @host, @port, @api_version, @security_protocol, extra_options)
+    @osh.connection_options = {:instrumentor => $fog_log}
+    begin
+      @swift ||= @osh.swift_service
+    rescue Excon::Errors::Unauthorized => err
+      logger.error("Access to Swift host #{@host} failed due to a bad username or password. #{err}")
+      msg = "Access to Swift host #{@host} failed due to a bad username or password. #{err}"
+      raise err, msg, err.backtrace
+    rescue => err
+      logger.error("Error connecting to Swift host #{@host}. #{err}")
+      msg = "Error connecting to Swift host #{@host}. #{err}"
+      raise err, msg, err.backtrace
+    end
+  end
+
+  def create_container
+    container = swift.directories.create(:key => container_name)
+    logger.debug("Swift container [#{container}] created")
+    container
+  rescue => err
+    logger.error("Error creating Swift container #{container_name}. #{err}")
+    msg = "Error creating Swift container #{container_name}. #{err}"
+    raise err, msg, err.backtrace
+  end
+
+  def download_single(source, destination)
+    object_key = @container_name
+    logger.debug("Downloading [#{source}] from bucket [#{bucket_name}] to local file [#{destination}]")
+
+    with_standard_s3_error_handling("downloading", source) do
+      if destination.kind_of?(IO)
+        s3.client.get_object(:bucket => bucket_name, :key => object_key) do |chunk|
```

Comment:
I don't think `s3` should be a thing in the `swift` code, right? Agree with @NickLaMuro, if this isn't implemented then let's raise a `NotImplementedError` rather than just copying and pasting unrelated code.